assets/src/edit-story/app/story/effects/useSaveMetaBoxes.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect, useRef, useState } from 'react';
+import { useFeature } from 'flagged';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../api';
+
+/**
+ * Function returning the current Meta Boxes DOM Node in the editor
+ * whether the meta box area is opened or not.
+ * If the MetaBox Area is visible returns it, and returns the original container instead.
+ *
+ * @param {string} location Meta Box location.
+ * @return {Element} Container
+ */
+function getMetaBoxContainer(location) {
+  // Class name as set in <MetaBoxesArea>
+  const area = document.querySelector(
+    `.web-stories-meta-boxes-area-${location} .metabox-location-${location}`
+  );
+  if (area) {
+    return area;
+  }
+
+  return document.querySelector('#metaboxes .metabox-location-' + location);
+}
+
+function usePrevious(value) {
+  const ref = useRef();
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+}
+
+function useSaveMetaBoxes({ story, isSaving, isAutoSaving }) {
```

Comment:
This effect hook is inspired by https://github.com/WordPress/gutenberg/blob/148e2b28d4cdd4465c4fe68d97fcee154a6b209a/packages/edit-post/src/store/effects.js#L24-L126

Reply:
Worth adding this a jsdoc block?